spec/controllers/order_management/order_details_controller_spec.rb

```diff
@@ -505,14 +516,41 @@ def reservation_params(reserve_start_at, actual_start_at = nil)
       end
     end
 
+    describe "when the price policy would change" do
+      let!(:previous_price_policy) { FactoryGirl.create(:item_price_policy, product: item, price_group: price_group, unit_cost: 19, start_date: 30.days.ago, expire_date: 28.days.ago) }
+      before { order_detail.backdate_to_complete!(29.days.ago) }
+
+      it "uses the fulfillment price policy rather than now's" do
+        @params[:order_detail] = {
+          account_id: new_account.id,
+        }
+        do_request
+        expect(order_detail.reload.price_policy).to eq(previous_price_policy)
+        expect(order_detail.actual_total).to eq(19)
+      end
+    end
+
     describe "changing quantity" do
       before do
+        order_detail.backdate_to_complete!(Time.current)
         @params[:order_detail] = { quantity: 2 }
       end
 
       it "updates the quanity" do
         expect { do_request }.to change { order_detail.reload.quantity }.to(2)
       end
+
+      it "updates the price while changing quantity" do
+        pending "currently buggy: it uses the price policy's calculation, not the params"
```

Comment:
Using `pending` was extra helpful because it showed me that my test was buggy (the order detail was not complete).